app/routes.py

```diff
@@ -1,2 +1,182 @@
+    elif sort == "desc":
+        model = model.query.order_by(model.title.desc())
+
+    return model
+
+
+@goals_bp.before_request
+@tasks_bp.before_request
+def get_model():
+
+    bps = {"tasks": (Task, "task"), "goals": (Goal, "goal")}
+    g.model, g.name = bps[request.blueprint]
+
+
+@goals_bp.route("", methods=["GET"])
+@tasks_bp.route("", methods=["GET"])
+def get_all():
+
+    model = g.model
+    if "sort" in request.args:
+        models = sort(model)
+    else:
+        models = model.query.all()
+
+    return jsonify([model.to_dict() for model in models])
+
+
+@goals_bp.route("/<id>", methods=["GET"])
+@tasks_bp.route("/<id>", methods=["GET"])
+def get_one(id):
+
+    model, name = g.model, g.name
+    model = validate(model, id)
+
+    return {f"{name}": model.to_dict()}
+
+
+@goals_bp.route("", methods=["POST"])
+@tasks_bp.route("", methods=["POST"])
+def create():
+
+    model, name = g.model, g.name
+    request_body = request.get_json()
+
+    try:
+        if model == Task:
+            new_entry = model(
+                title=request_body["title"],
+                description=request_body["description"],
+                completed_at=request_body["completed_at"],
+            )
+        elif model == Goal:
+            new_entry = model(title=request_body["title"])
+
+    except:
+        return {"details": "Invalid data"}, 400
+
+    db.session.add(new_entry)
+    db.session.commit()
+
+    return {f"{name}": new_entry.to_dict()}, 201
+
+
+@goals_bp.route("/<id>", methods=["PUT"])
+@tasks_bp.route("/<id>", methods=["PUT"])
+def update_one(id):
+    model, name = g.model, g.name
+    model = validate(model, id)
+    request_body = request.get_json()
+
+    model.update(request_body)
+
+    db.session.commit()
+    return {f"{name}": model.to_dict()}
+
+
+@goals_bp.route("/<id>", methods=["DELETE"])
+@tasks_bp.route("/<id>", methods=["DELETE"])
+def delete_one(id):
+
+    model, name = g.model, g.name
+    model = validate(model, id)
+
+    db.session.delete(model)
+    db.session.commit()
+
+    return {
+        "details": f'{name.capitalize()} {model.id} "{model.title}" successfully deleted'
+    }
+
+
+@tasks_bp.route("/<id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(id):
+
+    task = validate(Task, id)
+    text = f"Someone just completed the task {task.title}"
+    slack_bot(text)
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+
+    return {"task": task.to_dict()}
+
+
+@tasks_bp.route("/<id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(id):
+
+    task = validate(Task, id)
+    task.completed_at = None
+    db.session.commit()
+
+    return {"task": task.to_dict()}
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def add_tasks_to_goal(goal_id):
+    goal = validate(Goal, goal_id)
+    request_body = request.get_json()
+
+    try:
+        goal.tasks = [validate(Task, task_id) for task_id in request_body["task_ids"]]
```

Comment:
Nice use of `validate` within this list comprehension for clear and concise code.